server/controllers/finance/patientInvoice.js

```diff
@@ -227,6 +227,12 @@ function find(options) {
   // @FIXME Remove this with client side filter design
   delete options.patientNames;
 
+  let debtorJoin = ``;
+  const debtorKey = 'debtor_group_uuid';
```

Comment:
Are you sure this shouldn't be `debtor_uuid`?  Why filter on the `debtor_group_uuid`?

Reply:
I have not changed the join query to debtor. The only thing I tried to do is to add a flag that test is the user send ```debtor_group_uuid ``` in ```req.query```, if true : the join to debtor table is useful else we can skip it, that makes the request more faster.

Reply:
Ah, I understand now.  LGTM.